sdk/storage/azure-storage-blob/azure/storage/blob/_blob_client.py

```diff
@@ -2003,6 +2027,85 @@ def set_premium_page_blob_tier(self, premium_page_blob_tier, **kwargs):
         except StorageErrorException as error:
             process_storage_error(error)
 
+    def _set_blob_tags_options(self, blob_tags=None, **kwargs):
+        # type: (Optional[Dict[str, str]], **Any) -> Dict[str, Any]
+        headers = kwargs.pop('headers', {})
+
+        tags = serialize_blob_tags(blob_tags)
+
+        options = {
+            'tags': tags,
+            'version_id': kwargs.pop('version_id', None),
+            'timeout': kwargs.pop('timeout', None),
+            'cls': return_response_headers,
+            'headers': headers}
+        options.update(kwargs)
+        return options
+
+    @distributed_trace
+    def set_blob_tags(self, blob_tags=None, **kwargs):
+        # type: (Optional[Dict[str, str]], **Any) -> Dict[str, Any]
+        """The Set Tags operation enables users to set tags on a blob or specific blob version, but not snapshot.
+
+        :param blob_tags:
+            Blob tags
```

Comment:
Given that this is an optional parameter - I'm assuming the behaviour is the same as metadata?
I really don't like that it's optional.... but I guess we should be consistent... 
Either way - we should include a description about what happens if it's not specified, what the overwrite behaviour is etc

Reply:
Updated the doc, so if the tags are not specified they will be cleared.